sdk/core/azure-core/test/nlohmann-json-test/src/unit-msgpack.cpp

```diff
@@ -0,0 +1,1857 @@
+/*
+    __ _____ _____ _____
+ __|  |   __|     |   | |  JSON for Modern C++ (test suite)
+|  |  |__   |  |  | | | |  version 3.8.0
+|_____|_____|_____|_|___|  https://github.com/nlohmann/json
+
+Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+SPDX-License-Identifier: MIT
+Copyright (c) 2013-2019 Niels Lohmann <http://nlohmann.me>.
+
+Permission is hereby  granted, free of charge, to any  person obtaining a copy
+of this software and associated  documentation files (the "Software"), to deal
+in the Software  without restriction, including without  limitation the rights
+to  use, copy,  modify, merge,  publish, distribute,  sublicense, and/or  sell
+copies  of  the Software,  and  to  permit persons  to  whom  the Software  is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE  IS PROVIDED "AS  IS", WITHOUT WARRANTY  OF ANY KIND,  EXPRESS OR
+IMPLIED,  INCLUDING BUT  NOT  LIMITED TO  THE  WARRANTIES OF  MERCHANTABILITY,
+FITNESS FOR  A PARTICULAR PURPOSE AND  NONINFRINGEMENT. IN NO EVENT  SHALL THE
+AUTHORS  OR COPYRIGHT  HOLDERS  BE  LIABLE FOR  ANY  CLAIM,  DAMAGES OR  OTHER
+LIABILITY, WHETHER IN AN ACTION OF  CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE  OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "doctest_compatibility.h"
+
+#include <azure/core/internal/json.hpp>
+using Azure::Core::Internal::Json::json;
+
+#include <fstream>
+#include <iomanip>
+#include <set>
+#include <sstream>
+#include <test_data.hpp>
+
+namespace {
+class SaxCountdown {
+public:
+  explicit SaxCountdown(const int count) : events_left(count) {}
+
+  bool null() { return events_left-- > 0; }
+
+  bool boolean(bool) { return events_left-- > 0; }
+
+  bool number_integer(json::number_integer_t) { return events_left-- > 0; }
+
+  bool number_unsigned(json::number_unsigned_t) { return events_left-- > 0; }
+
+  bool number_float(json::number_float_t, const std::string&) { return events_left-- > 0; }
+
+  bool string(std::string&) { return events_left-- > 0; }
+
+  bool binary(std::vector<std::uint8_t>&) { return events_left-- > 0; }
+
+  bool start_object(std::size_t) { return events_left-- > 0; }
+
+  bool key(std::string&) { return events_left-- > 0; }
+
+  bool end_object() { return events_left-- > 0; }
+
+  bool start_array(std::size_t) { return events_left-- > 0; }
+
+  bool end_array() { return events_left-- > 0; }
+
+  bool parse_error(std::size_t, const std::string&, const json::exception&) { return false; }
+
+private:
+  int events_left = 0;
+};
+} // namespace
+
+TEST_CASE("MessagePack")
```

Comment:
Message pack is another thing we don't need tests for. It's a separate feature outside of JSON usage. Remove.

Reply:
Since you know much more about Json, I would like to sync with you some minutes after this is merged to review the list of test files and hear from you the ones we should also remove.
I will handle that as a new PR